dialogue-core/src/main/java/com/palantir/dialogue/core/BalancedNodeSelectionStrategyChannel.java

```diff
@@ -120,28 +121,34 @@
         for (SortableChannel channel : sortableChannels) {
             Optional<ListenableFuture<Response>> maybe = channel.delegate.maybeExecute(endpoint, request);
             if (maybe.isPresent()) {
+                log.info("Sending request to {}", channel.delegate.delegate);
                 if (rttSampler != null) {
                     rttSampler.maybeSampleRtts();
                 }
                 return maybe;
             }
+            log.info("Limited refused, on to the next one");
         }
 
         return Optional.empty();
     }
 
     private static SortableChannel[] computeScores(
-            @Nullable RttSampler rttSampler, List<MutableChannelWithStats> chans) {
+            @Nullable RttSampler rttSampler, List<MutableChannelWithStats> shuffledChannels) {
         // if the feature is disabled (i.e. RttSampling.DEFAULT_OFF), then we just consider every host to have a
         // rttSpectrum of '0'
-        float[] rttSpectrums = rttSampler != null ? rttSampler.computeRttSpectrums() : new float[chans.size()];
-
-        SortableChannel[] snapshotArray = new SortableChannel[chans.size()];
-        for (int i = 0; i < chans.size(); i++) {
-            MutableChannelWithStats channel = chans.get(i);
-            float rttSpectrum = rttSpectrums[i];
+        float[] rttSpectrums =
+                rttSampler != null ? rttSampler.computeRttSpectrums() : new float[shuffledChannels.size()];
+
+        SortableChannel[] snapshotArray = new SortableChannel[shuffledChannels.size()];
+        for (int i = 0; i < shuffledChannels.size(); i++) {
+            MutableChannelWithStats channel = shuffledChannels.get(i);
+            // beware rttSpectrums have the original order, but shuffledChannels are shuffled, so 'i' won't line up
+            float rttSpectrum = rttSpectrums[channel.hostIndex];
```

Comment:
this was the bug - changing i -> hostIndex made it work. Annoyingly since all the unit tests use a `getScoresForTesting` codepath, they never hit this.